src/server/system_services/system_store.js

```diff
@@ -658,125 +666,132 @@ class SystemStore extends EventEmitter {
      *
      */
     async make_changes(changes) {
```

Comment:
May I suggest that you make this change by adding a wrapping function like this:
```javascript
async make_changes(changes) {
    return this._load_serial.surround(async () => this._make_changes_internal(changes));
}

async _make_changes_internal(changes) {
// ... HERE COMES THE ORIGINAL CODE
```

Reply:
also easier for readability and history

Reply:
Ok, but it will make it harder to export all the values we need for the publishing load requests to the cluster.

Reply:
why not?

Reply:
ah I see because you publish outside the semaphore

Reply:
OK, still worth it I think that the internal method will return what's needed

Reply:
done

Reply:
Perhaps you can move back make_changes_in_background back to where it was? 
I share your desire to organize the code for reading, but we should minimize it to larger refactors.

Reply:
@guymguym this means to put this new method at the end of the file or else something will move

Reply:
no I meant just to move make_changes_in_background back to where it was to avoid a change on lines that are not required to change

Reply:
done